src/dynamic_load.cpp

```diff
@@ -2,20 +2,31 @@
 #include <iostream>
 #include "tracer.h"
 #include "tracer_factory.h"
-#include "version_check.h"
 
-int OpenTracingMakeTracerFactory(const char* opentracing_version, const void** error_category,
-                                 void** tracer_factory) try {
-  if (!datadog::opentracing::equal_or_higher_version(std::string(opentracing_version),
-                                                     std::string(OPENTRACING_VERSION))) {
+int OpenTracingMakeTracerFactoryFunction(const char* opentracing_version,
+                                         const char* opentracing_abi_version,
+                                         const void** error_category, void* error_message,
+                                         void** tracer_factory) try {
+  if (opentracing_version == nullptr || opentracing_abi_version == nullptr ||
+      error_message == nullptr || error_category == nullptr || tracer_factory == nullptr) {
+    std::cerr << "opentracing_version, opentracing_abi_version, error_message, `error_category, "
+                 " and tracer_factory must be non-null."
+              << std::endl;
+    std::terminate();
+  }
+
+  if (std::strcmp(opentracing_abi_version, OPENTRACING_ABI_VERSION) != 0) {
```

Comment:
The noise below should probably include ABI versions that weren't matching.

Reply:
Still not entirely certain if the old `opentracing_version` is important as well, but from other examples it doesn't seem to be worth checking